contracts/StaticOracle.sol

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+pragma abicoder v2;
+
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Factory.sol';
+import './interfaces/IStaticOracle.sol';
+import './libraries/PoolAddress.sol';
+import './libraries/OracleLibrary.sol';
+
+/// @title Uniswap V3 Static Oracle
+/// @notice Oracle contract price quoting against Uniswap V3 pools
+contract StaticOracle is IStaticOracle{
+
+    /// @inheritdoc IStaticOracle
+    address public override immutable factory;
+    /// @inheritdoc IStaticOracle
+    uint8 public immutable override cardinalityPerMinute;
+    uint24[] private knownFeeTiers;
+    uint32[4] private resistancePeriods;
+
+    constructor(address _factory, uint32[4] memory _resistancePeriods, uint8 _cardinalityPerMinute) {
+        factory = _factory;
+        resistancePeriods = _resistancePeriods;
+        cardinalityPerMinute = _cardinalityPerMinute;
+
+        // Assign default fee tiers
+        knownFeeTiers.push(300);
+        knownFeeTiers.push(5000);
+        knownFeeTiers.push(10000);
+    }
+
+    /// @inheritdoc IStaticOracle
+    function periodForResistanceLevel(ManipulationResistance resistance) public override view returns (uint32) {
+        return resistancePeriods[uint256(resistance)];
+    }    
+
+    /// @inheritdoc IStaticOracle
+    function supportedFeeTiers() external override view returns (uint24[] memory) {
+        return knownFeeTiers;
+    }
+
+    /// @inheritdoc IStaticOracle
+    function quote(
```

Comment:
Even though Solidity handles method overloading pretty well, js and other services (like etherscan) don't. And right now we have 6 `quote` functions, and 6 `prepare` functions. I think we should try to rename them so that they all have different names